src/SwitchImplIntf.ml

```diff
@@ -0,0 +1,59 @@
+open Import
+
+module Lsp : sig
+  val switch : LanguageClient.t -> string -> string option Promise.t
+end = struct
+  let switch client fileName =
+    let targetFileName : string option Promise.t =
+      LanguageClient.sendRequest client "ocaml/didSwitchImplIntf"
+        (fileName : string)
+        ()
+    in
+    targetFileName |> Promise.catch (fun _ -> Promise.return None)
```

Comment:
Better to annotate the type of `_` here. It's a bit dangerous to ignore a value of an unknown type.